### DISCUSSION TITLE
Pluggable Datasources

### DISCUSSION THREAD

**Discussion by santosh-d3vpl3x** [duckdb/duckdb#2566] [Q&A]
Is there any way we can add capability to connect to arbitrary data backend via plugin or extension system?

**Comment by Mytherin** [comment]
Yes, this is what the [Parquet extension](https://github.com/duckdb/duckdb/tree/master/extension/parquet) does, although only read operations and COPY TO are supported currently (i.e. you can't do insert, updates or deletes on that back-end yet). We are in the process of improving this setup to allow extensions to be installed from other sources (#2569).

**Comment by exaatto** [comment]
hi @santosh-d3vpl3x , I believe we are talking about the same feature. See this thread:

https://github.com/duckdb/duckdb/discussions/2728#discussion-3721029

I like the pg fdw way and think duckdb could borrow some ideas from that. Happy to see more of your ideas on that thread.
